src/header/common/referer.rs

```diff
@@ -0,0 +1,35 @@
+use header::{Header, HeaderFormat};
+use std::fmt::{self, Show};
+use header::shared::util::from_one_raw_str;
+
+/// The `Referer` header.
+///
+/// The Referer header is used by user agents to inform server about
+/// the page URL user has came from.
+///
+/// See alse [RFC 1945, section 10.13](http://tools.ietf.org/html/rfc1945#section-10.13).
+///
+/// Currently just a string, but maybe better replace it with url::Url or something like it.
```

Comment:
Is there need for more than just the Url?

On Sun, Jan 11, 2015, 12:29 PM Konstantin Stepanov <notifications@github.com>
wrote:

> In src/header/common/referer.rs
> <https://github.com/hyperium/hyper/pull/235#discussion-diff-22769809>:
>
> > @@ -0,0 +1,35 @@
> > +use header::{Header, HeaderFormat};
> > +use std::fmt::{self, Show};
> > +use header::shared::util::from_one_raw_str;
> > +
> > +/// The `Referer` header.
> > +///
> > +/// The Referer header is used by user agents to inform server about
> > +/// the page URL user has came from.
> > +///
> > +/// See alse [RFC 1945, section 10.13](http://tools.ietf.org/html/rfc1945#section-10.13).
> > +///
> > +/// Currently just a string, but maybe better replace it with url::Url or something like it.
>
> But then headers must have access not only to a request, but to a response
> too. We need to generalize them if we want to do it this way, for example
> create HttpContext trait with common methods and implement it for both
> HttpRequest and HttpResponse.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hyperium/hyper/pull/235/files#r22769809>.
>